src/js/tls.js

```diff
@@ -0,0 +1,190 @@
+/* Copyright 2018-present Samsung Electronics Co., Ltd. and other contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+var net = require('net');
+var util = require('util');
+
+function Tls() {
+}
+
+Tls.Server = function(options) {
+  return net.Server(options);
+}
+
+Tls.Server.prototype.addContext = function(hostname, context) {
+  if (!util.isString(hostname)) {
+    throw new TypeError('hostname must be a string');
+  }
+
+  if (!util.isObject(context)) {
+    throw new TypeError('context must be an object');
+  }
+}
+
+Tls.Server.prototype.address = function() {
+}
+
+Tls.Server.prototype.close = function(callback) {
+  if (callback && !util.isFunction(callback)) {
+    throw new TypeError('callback must be a function');
+  }
+}
+
+Tls.Server.prototype.getTicketKeys = function() {
+}
+
+Tls.Server.prototype.listen = function() {
+}
+
+Tls.Server.prototype.setTicketKeys = function(keys) {
+  if (!util.isBuffer(keys)) {
+    throw new TypeError('keys must be a buffer');
+  }
+}
+
+
+Tls.TLSSocket = function(socket, opts) {
+  var _socket = socket;
+
+  if (!_socket || !(socket instanceof net.Socket)) {
+    _socket = new net.Socket();
+  }
+
+  this.encrypted = true;
+  this.isServer = !!opts.isServer || false;
+  this.requestCert = !!opts.requestCert || false;
+
+  if (opts.NPNProtocols && !util.isBuffer(opts.NPNProtocols)) {
+    throw new TypeError('TLSSocket - options.NPNProtocols must be a buffer.');
+  }
+
+  if (opts.ALPNProtocols && !util.isBuffer(opts.ALPNProtocols)) {
+    throw new TypeError('TLSSocket - options.ALPNProtocols must be a buffer.');
+  }
+
+  if (opts.SNICallback && !util.isFunction(opts.SNICallback)) {
+    throw new TypeError('TLSSocket - options.SNICallback must be a function.');
+  }
+
+  if (opts.session && !util.isBuffer(opts.session)) {
+    throw new TypeError('TLSSocket - options.session should be a buffer.');
+  }
+
+  return this;
+}
+
+Tls.TLSSocket.prototype.address = function() {
+}
+
+Tls.TLSSocket.prototype.disableRenegotiation = function() {
+  this.disableRenegotiation = true;
+}
+
+Tls.TLSSocket.prototype.encrypted = function() {
+  return this.encrypted;
+}
+
+Tls.TLSSocket.prototype.getCipher = function() {
+}
```

Comment:
What about throwing `new TypeError("Unimplemented");` in every unimplemented functions?

Reply:
Great idea!